clang/tools/clang-scan-deps/ClangScanDeps.cpp

```diff
@@ -519,6 +528,8 @@ class FullDeps {
                       });
                     JOS.attributeArray("clang-module-deps",
                                        toJSONSorted(JOS, I.ClangModuleDeps));
+                    JOS.attributeArray("visible-clang-modules",
```

Comment:
I'd suggest adding a flag to `clang-scan-deps` to include this information only if it was explicitly requested. That would reduce churn in unrelated test files and keep them focused on asserting what they actually care about.

Reply:
I was thinking about it from the perspective of ensuring the information is always correct, it's useful to always emit, so I wanted to add it to tests that were already checking all, or most, of the output. Do you think it just adds noise?

Reply:
FWIW we've historically not been great at this - for example lots of tests check the set of input files even though they only really want to check the command-line. I've typically done this so that `FileCheck` can run line-by-line via `CHECK-NEXT` in order to avoid some sharp edges with using `CHECK` in a structured input. This has bitten me later, for example when making sure all input paths we report are absolute and normalized, which has caused lots of unnecessary churn. Looking back, I think we should've introduced a flag to `clang-scan-deps` to skip outputting the input files and simplify most of our tests that don't care about input files.

Reply:
We agreed offline that a good path forward is to add the optional flag now so that it doesn't influence other tests.  I would still like this info to be printed by default, so that is now blocked on adding a filter option for tests to be able to specify which fields to print/check for.